instant-xml/src/lib.rs

```diff
@@ -10,15 +10,165 @@ pub use macros::{FromXml, ToXml};
 pub mod parse;
 
 pub trait ToXml {
-    fn write_xml<W: fmt::Write>(&self, write: &mut W) -> Result<(), Error>;
-
     fn to_xml(&self) -> Result<String, Error> {
-        let mut out = String::new();
-        self.write_xml(&mut out)?;
-        Ok(out)
+        let mut parent_prefixes = BTreeSet::new();
+        let mut output = String::new();
+        let mut serializer = Serializer {
+            parent_prefixes: &mut parent_prefixes,
+            output: &mut output,
+        };
+        self.serialize(&mut serializer, None)?;
+        Ok(output)
+    }
+
+    fn serialize(
+        &self,
+        serializer: &mut Serializer,
+        field_data: Option<&mut FieldData>,
+    ) -> Result<(), Error>;
+}
+
+macro_rules! to_xml_for_number {
+    ($typ:ty) => {
+        impl ToXml for $typ {
+            fn serialize(
+                &self,
+                serializer: &mut Serializer,
+                field_data: Option<&mut FieldData>,
+            ) -> Result<(), Error> {
+                let field_data = match field_data {
```

Comment:
I think we should return an error in the case a scalar type gets called with `None` `FieldData`.

Reply:
It was meant to be used for tuple structs, but I don't know if we are planning to support them.

Reply:
We should probably support tuple structs, but not with scalars (directly) inside them.